devOpcuaSup/open62541/DataElementOpen62541.h

```diff
@@ -135,15 +135,15 @@ inline bool isWithinRange<UA_Int64, epicsUInt64> (const epicsUInt64 &value) {
     return !(value > static_cast<UA_UInt64>(std::numeric_limits<UA_Int64>::max()));
 }
 
-template<>
-inline bool isWithinRange<epicsInt32, UA_UInt64> (const UA_UInt64 &value) {
-    return !(value > static_cast<UA_UInt64>(std::numeric_limits<epicsInt32>::max()));
-}
+//template<>                                                                                                      // same as bool isWithinRange<UA_Uint32, epicsUInt64>(const epicsUInt64 &value)
+//inline bool isWithinRange<epicsInt32, UA_UInt64> (const UA_UInt64 &value) {
+//    return !(value > static_cast<UA_UInt64>(std::numeric_limits<epicsInt32>::max()));
+//}
 
-template<>
-inline bool isWithinRange<epicsInt64, UA_UInt64> (const UA_UInt64 &value) {
-    return !(value > static_cast<UA_UInt64>(std::numeric_limits<epicsInt64>::max()));
-}
+//template<>                                                                                                      // same as bool isWithinRange<UA_Uint64, epicsUInt64>(const epicsUInt64 &value)
+//inline bool isWithinRange<epicsInt64, UA_UInt64> (const UA_UInt64 &value) {
+//    return !(value > static_cast<UA_UInt64>(std::numeric_limits<epicsInt64>::max()));
+//}
```

Comment:
It is not the same -- on 64 bit Linux at least. Removing those gives this warning with gcc 6.2.1
In file included from ../../devOpcuaSup/open62541/DataElementOpen62541.cpp:29:0:
../../devOpcuaSup/open62541/DataElementOpen62541.h: In instantiation of ‘bool DevOpcua::isWithinRange(const FROM&) [with TO = int; FROM = long unsigned int]’:
../../devOpcuaSup/open62541/DataElementOpen62541.h:838:61:   required from ‘long int DevOpcua::DataElementOpen62541::readScalar(ET*, dbCommon*, DevOpcua::ProcessReason*, epicsUInt32*, char*, epicsUInt32) [with ET = int; dbCommon = dbCommon; epicsUInt32 = unsigned int]’
../../devOpcuaSup/open62541/DataElementOpen62541.cpp:444:97:   required from here
../../devOpcuaSup/open62541/DataElementOpen62541.h:99:20: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     return !(value < std::numeric_limits<TO>::lowest() || value > std::numeric_limits<TO>::max());
              ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../devOpcuaSup/open62541/DataElementOpen62541.h: In instantiation of ‘bool DevOpcua::isWithinRange(const FROM&) [with TO = long long int; FROM = long unsigned int]’:
../../devOpcuaSup/open62541/DataElementOpen62541.h:838:61:   required from ‘long int DevOpcua::DataElementOpen62541::readScalar(ET*, dbCommon*, DevOpcua::ProcessReason*, epicsUInt32*, char*, epicsUInt32) [with ET = long long int; dbCommon = dbCommon; epicsUInt32 = unsigned int]’
../../devOpcuaSup/open62541/DataElementOpen62541.cpp:455:97:   required from here
../../devOpcuaSup/open62541/DataElementOpen62541.h:99:20: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]

Reply:
Both the EPICS types and the UA types are explicit. There should be no ambiguity.